openwisp_users/tests/test_api/test_api.py

```diff
@@ -49,7 +49,7 @@ def test_organization_list_nonsuperuser_api(self):
     def test_organization_post_api(self):
         path = reverse('users:organization_list')
         data = {'name': 'test-org', 'slug': 'test-org'}
-        with self.assertNumQueries(6):
+        with self.assertNumQueries(4):
```

Comment:
I also do not understand why the queries go down, do you have an idea of why this is happening?

Reply:
After looking deep into the code, I found that in the serializer `validate()` method we are not calling `super().validate()`; therefore, we got 2 queries less than before.

#### Before (with slug `UniqueValidator`):
- `UniqueValidator()` and `super().validate()` both are executing similar queries (total 6 queries).

```sql
2. SELECT (1) AS "a" FROM "openwisp_users_organization" WHERE "openwisp_users_organization"."slug" = 'test-org' LIMIT 1
3. SELECT (1) AS "a" FROM "openwisp_users_organization" WHERE "openwisp_users_organization"."slug" = 'test-org' LIMIT 1
```
![Screenshot from 2023-01-12 16-05-39](https://user-images.githubusercontent.com/56113566/212048896-8cacc7bf-276c-4170-9386-0800efcbd6d9.png)

#### After (with custom slug validation)

- First we validate `slug_field` and then `super().validate()` (total 6 queries).

![Screenshot from 2023-01-12 16-16-13](https://user-images.githubusercontent.com/56113566/212049217-cc6ec468-1d65-4d54-be90-88d931bdc14b.png)